config/initializers/hyrax-doi.rb

```diff
@@ -168,3 +168,25 @@ def assign_for(object:, attribute: :identifier)
     object
   end
 end
+
+# Forces DataCiteRegistrar to initializew a new client everytime. 
+# A fix for #245 https://github.com/scientist-softserv/britishlibrary/issues/245
+# Prevents leaching of datacite config between tenants when minting DOIs by ensuring
+# that the datacite credentials always come from the account that kicked the job off
+Hyrax::DOI::DataCiteRegistrar.class_eval do
+
+  def initialize(builder: Hyrax::Identifier::Builder.new(prefix: self.prefix))
```

Comment:
One foible with this is that the `self.prefix` at this point of evaluation is the class's assigned prefix (which is set during the request.  It is not the data `datacite_endpoint.prefix`.

I suspect we would want to do the following:

```ruby
def initialize(builder: nil)
...
super(builder: Hyrax::Identifier::Builder.new(prefix: self.prefix))
```